From f99c29c8257ca24a0cdb8c1484c80d314bc84e25 Mon Sep 17 00:00:00 2001
From: Juan Quintela <quintela@redhat.com>
Date: Thu, 4 Mar 2010 23:11:25 -0300
Subject: [PATCH 03/42] slirp/misc.c: fix warning with _FORTIFY_SOURCE

RH-Author: Juan Quintela <quintela@redhat.com>
Message-id: <884738b88c69a99698993010a268fa23f961aa61.1267743950.git.quintela@redhat.com>
Patchwork-id: 7533
O-Subject: [PATCH 03/32] slirp/misc.c: fix warning with _FORTIFY_SOURCE
Bugzilla: 567099
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>
RH-Acked-by: Amit Shah <amit.shah@redhat.com>
RH-Acked-by: Marcelo Tosatti <mtosatti@redhat.com>

From: Kirill A. Shutemov <kirill@shutemov.name>

CC    slirp/misc.o
cc1: warnings being treated as errors
slirp/misc.c: In function 'fork_exec':
slirp/misc.c:209: error: ignoring return value of 'write', declared with attribute warn_unused_result
make: *** [slirp/misc.o] Error 1

Signed-off-by: Kirill A. Shutemov <kirill@shutemov.name>
Signed-off-by: Blue Swirl <blauwirbel@gmail.com>
(cherry picked from commit f0d98b0517097509e1ebe2ac333f1ed70e7ea4f1)

Signed-off-by: Juan Quintela <quintela@redhat.com>
---
 slirp/misc.c |   10 ++--------
 1 files changed, 2 insertions(+), 8 deletions(-)

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 slirp/misc.c |   10 ++--------
 1 files changed, 2 insertions(+), 8 deletions(-)

diff --git a/slirp/misc.c b/slirp/misc.c
index c76ad8f..05f4fb3 100644
--- a/slirp/misc.c
+++ b/slirp/misc.c
@@ -200,14 +200,8 @@ fork_exec(struct socket *so, const char *ex, int do_pty)
 		execvp(argv[0], (char **)argv);
 
 		/* Ooops, failed, let's tell the user why */
-		  {
-			  char buff[256];
-
-			  snprintf(buff, sizeof(buff),
-                                   "Error: execvp of %s failed: %s\n",
-                                   argv[0], strerror(errno));
-			  write(2, buff, strlen(buff)+1);
-		  }
+        fprintf(stderr, "Error: execvp of %s failed: %s\n",
+                argv[0], strerror(errno));
 		close(0); close(1); close(2); /* XXX */
 		exit(1);
 
-- 
1.6.3.rc4.29.g8146

